src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -159,7 +159,9 @@ private Future<Response> postSamlLogin(SamlLoginRequest requestEntity, RoutingCo
 
     // register non-persistent session (this request only) to overWrite relayState
     Session session = new SharedDataSessionImpl(new PRNG(vertxContext.owner()));
-    session.put(SAML_RELAY_STATE_ATTRIBUTE, relayState);
+    // csrfToken without url because RelayState data MUST NOT exceed 80 bytes in length:
+    // https://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf
+    session.put(SAML_RELAY_STATE_ATTRIBUTE, csrfToken);
```

Comment:
Since this shortening seems to be limited to what is stored here in the session, and then checked against the cookie, it seems like the IdP metadata is not in scope and therefore there's no update of metadata necessary. @julianladisch  is that your understanding as well?